src/Configurator/DockerComposeConfigurator.php

```diff
@@ -125,6 +138,45 @@ private function normalizeConfig(array $config): array
         return $config;
     }
 
+    /**
+     * Finds the Docker Compose file according to these rules: https://docs.docker.com/compose/reference/envvars/#compose_file.
+     */
+    private function findDockerComposeFile(string $rootDir, string $file): ?string
+    {
+        if (isset($_SERVER['COMPOSE_FILE'])) {
+            $files = explode('\\' === \DIRECTORY_SEPARATOR ? ';' : ':', $_SERVER['COMPOSE_FILE']);
+            foreach ($files as $f) {
+                if ($file !== basename($f)) {
+                    continue;
+                }
+
+                if (!$this->filesystem->isAbsolutePath($f)) {
+                    $f = realpath(sprintf('%s/%s', $rootDir, $f));
+                }
+
+                return $this->filesystem->exists($f) ? $f : null;
+            }
+        }
+
+        // COMPOSE_FILE not set, or doesn't contain the file we're looking for
+        $dir = $rootDir;
+        $previousDir = null;
+        do {
+            // Test with the ".yaml" extension if the file doesn't end up with ".yml".
+            if (
+                file_exists($dockerComposeFile = sprintf('%s/%s', $dir, $file)) ||
```

Comment:
`file_exists` versus `filesystem->exists` at line 157?

Reply:
Changed every occurrences to use `Filesystem` for consistency.